analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -120,13 +120,24 @@ class Gradle(
     private val dependencyHandler = GradleDependencyHandler(managerName, maven)
     private val graphBuilder = DependencyGraphBuilder(dependencyHandler)
 
+    // The path to the root project. In a single-project, just points to the project path. Is only null when not yet
+    // initialized.
+    private var currentRootProjectDir: File? = null
+
     override fun createPackageManagerResult(projectResults: Map<File, List<ProjectAnalyzerResult>>) =
         PackageManagerResult(projectResults, graphBuilder.build(), graphBuilder.packages())
 
     override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> {
         val gradleSystemProperties = mutableListOf<Pair<String, String>>()
         val gradleProperties = mutableListOf<Pair<String, String>>()
 
+        val projectDir = definitionFile.parentFile
+        val isRootProject = GRADLE_SETTINGS_FILES.any { projectDir.resolve(it).isFile }
+        val isIndependentProject = currentRootProjectDir?.let { projectDir.relativeToOrNull(it) } == null
```

Comment:
According to the documentation of [relativeToOrNull](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/relative-to-or-null.html) it returns a File with an empty path if `projectDir` matches the base file (in this case `it`).
This would be the case for an independent project and evaluate to false with `== null`.

Reply:
> Wouldn't this be the case for an independent project and evaluate to false with `== null`?

I don't think so. Before this line, it should never be the case that `currentRootProjectDir == projectDir` because:

1. `currentRootProjectDir` might be `null`, but `projectDir` can't be `null`
2. `currentRootProjectDir` might point to the project directory of the previously resolved project, but `projectDir` always points to the directory of the current project to be resolved

Reply:
Somewhat unrelated, but I could also rewrite this to

    val isIndependentProject = currentRootProjectDir?.let { projectDir.startsWith(it) } != true

Is that any better?

Reply:
Somewhat unrelated, but I could also rewrite this to

    val isIndependentProject = currentRootProjectDir?.let { projectDir.startsWith(it) } != true

Is that any better?

Reply:
Yes I think that is clearer.